common/reflect_resource.go

```diff
@@ -91,12 +93,13 @@ func getJsonFieldName(typeField reflect.StructField) string {
 // SchemaPath helps to navigate
 func SchemaPath(s map[string]*schema.Schema, path ...string) (*schema.Schema, error) {
 	cs := s
-	for _, p := range path {
+	for i := 0; i < len(path); i++ {
+		p := path[i]
 		v, ok := cs[p]
 		if !ok {
 			return nil, fmt.Errorf("missing key %s", p)
 		}
-		if p == path[len(path)-1] {
+		if i == len(path)-1 {
```

Comment:
There was a bug with the previous way of implementation because you could have an item in the path that is the same string as the last item, which would make this function return early.